File: internal/apiclient/chats.go
package apiclient

import (
	"context"
	"net/url"

	"github.com/jpnorenam/rag-snap/internal/chatstore"
)

// ListChats returns saved chat summaries newest-first. A non-empty search filters
// server-side by case-insensitive substring over title and transcript content.
func (c *Client) ListChats(ctx context.Context, search string) ([]chatstore.Summary, error) {
	path := "/1.0/chats"
	if search != "" {
		path += "?search=" + url.QueryEscape(search)
	}
	var out []chatstore.Summary
	if err := c.Sync(ctx, "GET", path, nil, &out); err != nil {
		return nil, err
	}
	return out, nil
}

// GetChat returns a full saved chat, including its transcript.
func (c *Client) GetChat(ctx context.Context, id string) (*chatstore.Chat, error) {
	var chat chatstore.Chat
	if err := c.Sync(ctx, "GET", "/1.0/chats/"+id, nil, &chat); err != nil {
		return nil, err
	}
	return &chat, nil
}

// DeleteChat removes a saved chat.
func (c *Client) DeleteChat(ctx context.Context, id string) error {
	return c.Sync(ctx, "DELETE", "/1.0/chats/"+id, nil, nil)
}

File: ui/lib/api/chats.ts
import { getSync, deleteSync } from "./envelope";

// ChatTurn is one exchange in a saved conversation, mirroring the daemon's
// chatstore.Turn.
export interface ChatTurn {
  role: "user" | "assistant";
  content: string;
}

// ChatSummary is the transcript-free view of a saved chat from GET /1.0/chats.
export interface ChatSummary {
  id: string;
  title: string;
  created_at: string;
  updated_at: string;
  model?: string;
  bases?: string[] | null;
  turn_count: number;
}

// SavedChat is the full saved chat from GET /1.0/chats/{id}.
export interface SavedChat extends ChatSummary {
  turns: ChatTurn[];
}

// listChats returns saved chat summaries newest-first. A non-empty search filters
// server-side by case-insensitive substring over title and transcript content.
export async function listChats(search?: string): Promise<ChatSummary[]> {
  const path = search ? `/1.0/chats?search=${encodeURIComponent(search)}` : "/1.0/chats";
  const chats = await getSync<ChatSummary[] | null>(path);
  return chats ?? [];
}

// getChat returns a full saved chat including its transcript.
export async function getChat(id: string): Promise<SavedChat> {
  return getSync<SavedChat>(`/1.0/chats/${encodeURIComponent(id)}`);
}

// deleteChat removes a saved chat.
export async function deleteChat(id: string): Promise<void> {
  await deleteSync(`/1.0/chats/${encodeURIComponent(id)}`);
}
